lib/assets/javascripts/turbograft/turbohead.coffee

```diff
@@ -4,39 +4,56 @@ class window.TurboHead
   update: (successCallback, failureCallback) ->
     activeAssets = extractTrackedAssets(@activeDocument)
     upstreamAssets = extractTrackedAssets(@upstreamDocument)
-    {activeScripts, newScripts} = processScripts(activeAssets, upstreamAssets)
 
-    if hasScriptConflict(activeScripts, newScripts)
-      return failureCallback()
+    newScripts = upstreamAssets
+      .filter(filterForNodeType('SCRIPT'))
+      .filter(noMatchFor({attribute: 'src', inCollection: activeAssets}))
 
-    updateLinkTags(activeAssets, upstreamAssets)
-    updateScriptTags(@activeDocument, newScripts, successCallback)
+    newLinks = upstreamAssets
+      .filter(filterForNodeType('LINK'))
+      .filter(noMatchFor({attribute: 'href', inCollection: activeAssets}))
 
-updateLinkTags = (activeAssets, upstreamAssets) ->
-  activeLinks = activeAssets.filter(filterForNodeType('LINK'))
-  upstreamLinks = upstreamAssets.filter(filterForNodeType('LINK'))
-  remainingActiveLinks = removeStaleLinks(activeLinks, upstreamLinks)
-  reorderedActiveLinks = reorderActiveLinks(remainingActiveLinks, upstreamLinks)
-  insertNewLinks(reorderedActiveLinks, upstreamLinks)
+    if newScripts.concat(newLinks).some(hasAssetConflicts(activeAssets))
+      return failureCallback()
```

Comment:
`TurboHead`'s API should be:
```
constructor(activeDocument, upstreamDocument)
boolean hasUpdatedAsset()
insertNewAssets(callback)
```

Then this oddly named var won't be necessary, and the [bizzaro variable creation to simulate named parameters](https://github.com/Shopify/turbograft/blob/bd66d218af5b438baeb3f73ff1ae07f26d342ace/lib/assets/javascripts/turbograft/turbolinks.coffee#L139) code won't be necessary.